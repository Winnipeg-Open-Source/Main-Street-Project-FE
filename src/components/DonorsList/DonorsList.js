import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'pcln-design-system';
import List from 'components/List';
import DonorCard from 'components/DonorCard';

function DonorsList ({ className, donors }) {
    return (
        <List className={className}>
            {donors && donors.length > 0
                ? donors.map(donor => <DonorCard key={donor.id} {...donor} />)
                : <Text textAlign='center'>No donors yet...</Text>
            }
        </List>
    );
}

DonorsList.displayName = 'DonorsList';

DonorsList.propTypes = {
    className: PropTypes.string,
    donors: PropTypes.array,
};

DonorsList.defaultProps = {
    className: '',
};

export default DonorsList;
