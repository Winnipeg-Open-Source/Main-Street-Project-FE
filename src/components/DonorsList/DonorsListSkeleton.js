import React, { memo } from 'react';
import PropTypes from 'prop-types';
import List from 'components/List';
import DonorCard from 'components/DonorCard';

function DonorCardSkeleton ({ className }) {
    return (
        <List className={className}>
            <DonorCard isLoading />
            <DonorCard isLoading />
            <DonorCard isLoading />
        </List>
    );
}

DonorCardSkeleton.displayName = 'DonorCardSkeleton';

DonorCardSkeleton.propTypes = {
    className: PropTypes.string,
};

DonorCardSkeleton.defaultProps = {
    className: '',
};

export default memo(DonorCardSkeleton);
