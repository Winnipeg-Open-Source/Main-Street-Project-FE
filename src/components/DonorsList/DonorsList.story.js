import React from 'react';
import DonorsList from './DonorsList';

const singleDonor = [
    {
        id: 1,
        name: 'Costco',
        lastDonationDate: '2019 December 20',
    },
];

const multipleDonors = [
    {
        id: 1,
        name: 'Costco',
        lastDonationDate: '2019 December 20',
    },
    {
        id: 2,
        name: 'Sobeys',
        lastDonationDate: '2019 December 21',
    },
];

export const NoDonors = () => <DonorsList />;

export const SingleDonor = () => <DonorsList donors={singleDonor} />;

export const MultipleDonors = () => <DonorsList donors={multipleDonors} />;

export default {
    title: 'DonorsList',
    component: DonorsList,
};
