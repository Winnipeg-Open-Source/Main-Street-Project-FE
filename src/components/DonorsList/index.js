import WithSkeleton from 'hocs/WithSkeleton';
import DonorsList from './DonorsList';
import DonorsListSkeleton from './DonorsListSkeleton';

export default WithSkeleton(DonorsListSkeleton, DonorsList);
