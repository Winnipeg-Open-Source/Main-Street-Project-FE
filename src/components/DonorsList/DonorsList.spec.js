import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import DonorsList from './DonorsList';
import DonorsListSkeleton from './DonorsListSkeleton';

const singleDonorProps = {
    donors: [
        {
            id: 1,
            name: 'Costco',
            lastDonationDate: '2019 December 20',
        },
    ],
};

const multipleDonorsProps = {
    donors: [
        {
            id: 1,
            name: 'Costco',
            lastDonationDate: '2019 December 20',
        },
        {
            id: 2,
            name: 'Sobeys',
            lastDonationDate: '2019 December 21',
        },
    ],
};

describe('DonorsList', () => {
    it('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<DonorsList />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('skeleton renders without crashing', () => {
        const { asFragment } = renderWithTheme(<DonorsListSkeleton />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('has no donors', () => {
        const { queryByText } = renderWithTheme(<DonorsList />);

        const noDonorsText = queryByText('No donors yet...');
        const title = queryByText(/Costco/i);

        expect(noDonorsText).toBeInTheDocument();
        expect(title).not.toBeInTheDocument();
    });

    it('has single donor', () => {
        const { queryByText } = renderWithTheme(
            <DonorsList {...singleDonorProps} />
        );

        const noDonorsText = queryByText('No donors yet...');
        const title = queryByText(/Costco/i);
        const secondTitle = queryByText(/Sobeys/i);

        expect(noDonorsText).not.toBeInTheDocument();
        expect(title).toBeInTheDocument();
        expect(secondTitle).not.toBeInTheDocument();
    });

    it('has multiple donors', () => {
        const { queryByText } = renderWithTheme(
            <DonorsList {...multipleDonorsProps} />
        );

        const noDonorsText = queryByText('No donors yet...');
        const title = queryByText(/Costco/i);
        const secondTitle = queryByText(/Sobeys/i);

        expect(noDonorsText).not.toBeInTheDocument();
        expect(title).toBeInTheDocument();
        expect(secondTitle).toBeInTheDocument();
    });
});
