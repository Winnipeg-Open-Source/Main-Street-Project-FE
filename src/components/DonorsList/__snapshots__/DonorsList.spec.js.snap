// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonorsList renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-LzLMl bRtbEd sc-LzLMm jaRkvP"
  >
    <div
      class="sc-AykKC sc-AykKE sc-LzLMn bcymSi"
      width="1"
    >
      <div
        class="sc-AykKF cuPuLz"
      >
        No donors yet...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
